### PR TITLE
Set Material design default side bar widths.

### PR DESCRIPTION
**Pull request by zgosalvez**
Based on [Side nav structure](https://www.google.com/design/spec/layout/structure.html#structure-side-nav).

### CODE REVIEW

File: Sources/SideNavigationController.swift [hunk 1]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
+    @IBInspectable public private(set) var rightViewWidth: CGFloat! {
+        get {
+            if let rightViewWidth = _rightViewWidth {
+                return rightViewWidth
+            }
+            
+            return MaterialDevice.width
+        }
+        set {
+            _leftViewWidth = newValue
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: If you set the value here, the left side is updated, not right. Is this your intention?

Comment (zgosalvez) [inline Sources/SideNavigationController.swift]: My bad. Let me create a new PR.

File: Sources/SideNavigationController.swift [hunk 2]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
-	/**
-	A CGFloat property to access the width that the rightView
-	opens up to.
-	*/
-	@IBInspectable public private(set) var rightViewWidth: CGFloat = 240
+    /**
+     A CGFloat property to access the width that the leftView
+     opens up to.
+     */
+    @IBInspectable public private(set) var leftViewWidth: CGFloat! {
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: What happens on devices that are 414 in width like a 6 plus?

File: Sources/SideNavigationController.swift (hunk 1 above)
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: wait! let me finish the review.

File: Sources/SideNavigationController.swift (hunk 2 above)
Comment (zgosalvez) [inline Sources/SideNavigationController.swift]: Left sidebar width stops at 320, as per Material docs.

File: Sources/SideNavigationController.swift [hunk 3]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
+        }
+    }
+    
+    private var _leftViewWidth: CGFloat?
+    
+    /**
+     A CGFloat property to access the width that the rightView
+     opens up to.
+     */
+    @IBInspectable public private(set) var rightViewWidth: CGFloat! {
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: Why not set the `rightThreshold` which creates the right max width, and thus not hardcoded to 320, which is set in the method. What if someone wants an expanded view in an iPad, that spans to let's say 600 px when in landscape mode?

File: Sources/SideNavigationController.swift (hunk 1 above)
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: Well the 320 max is a device thing, and we have had requests to allow people to set the max width. It is easy to use the `setLeftViewWidth` and `setRightViewWidth` that can easily be set to 320. As an animation and graphics framework, this creates a problem. Not all users who use Material use it for Material Design, they use it to customize the components. This seems like an unnecessary limitation when one can accomplish the same goal without the framework being limited.

Comment (zgosalvez) [inline Sources/SideNavigationController.swift]: Not sure how it's going to be a limitation if users can still customise it using the said methods. The reason for this pull request is to set the defaults to Material design guidelines. I think it makes more sense if users get that out of the box, then if desired, customise the components themselves; They're guidelines after all.

File: Sources/SideNavigationController.swift [hunk 4]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
-	opens up to.
-	*/
-	@IBInspectable public private(set) var rightViewWidth: CGFloat = 240
+    /**
+     A CGFloat property to access the width that the leftView
+     opens up to.
+     */
+    @IBInspectable public private(set) var leftViewWidth: CGFloat! {
+        get {
+            if let leftViewWidth = _leftViewWidth {
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: Here you are not following the codebase style. The type should be set.

File: Sources/SideNavigationController.swift [hunk 5]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
+                return leftViewWidth
+            }
+            
+            let maximumWidth = CGFloat(320)
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: rather than CGFloat(320), you can type set and do let maximumWidth: CGFloat = 320.

Comment (danieldahan) [inline Sources/SideNavigationController.swift]: Which also follows the style of the rest of the codebase.

File: Sources/SideNavigationController.swift [hunk 6]
```diff
@@ -276,17 +276,50 @@ public class SideNavigationController : UIViewController, UIGestureRecognizerDel
...
+     A CGFloat property to access the width that the rightView
+     opens up to.
+     */
+    @IBInspectable public private(set) var rightViewWidth: CGFloat! {
+        get {
+            if let rightViewWidth = _rightViewWidth {
+                return rightViewWidth
+            }
+            
+            return MaterialDevice.width
```
Comment (danieldahan) [inline Sources/SideNavigationController.swift]: So rightViewWidth is set to MaterialDevice.width if the property is nil? The animation will then extend that to the device width, is this the desired effect?

Comment (zgosalvez) [inline Sources/SideNavigationController.swift]: Yes.

> Maximum width applies only when using a left nav. When using a right nav, the panel can cover the full width of the screen.

File: Sources/SideNavigationController.swift (hunk 5 above)
Comment (zgosalvez) [inline Sources/SideNavigationController.swift]: Just curious, any particular reason for this code style?

Comment (danieldahan) [inline Sources/SideNavigationController.swift]: Yes, I like explicit, very clear, unambiguous code, when achievable. At this point, the style is very important, as no one has convinced me how to approach it. Swift is a young language trying to be shaped by many people. Its better to code clearly than adopt a style that may change. Coming from c and c++ background, being explicit is important. If swift is type safe, it should be written type safely. As well I am javascript heavy, and swifts remind me of javascript and c++ combined. So i like free flowing languages as well. In this case, I chose to be very explicit.

**Comment by danieldahan** [comment]
I am going to look at this in the morning. Give it a fresh look. I am not opposed to your suggestion, mainly, I want to encourage a cohesive approach.

**Comment by danieldahan** [comment]
First, I really appreciate the effort on this PR. I agree that the default width was not accurate and needed to be updated. Based on Material Design's drawer spec:

> Side nav width: Equal to the screen width minus the height of the action bar. In the example shown this is 56dp from the right edge of the screen. The maximum width of the nav drawer is five times the standard increment (56dp on mobile and 64dp on tablet).

I have defaulted the width to 280 for mobile, and 320 for tablet. Where I had issues with the PR is:

1. Didn't follow coding style. 
2. The `setLeftViewWidth` and `setRightViewWidth` were not handled correctly with the change. 
3. I love the default, but I don't like the maximum width limitation - as someone will ask, can I set the width to more than 280 or 320. Defaults are nice, and specs are great, but hardcoded rules in this situation feels limiting. 

In general, I want to follow the spec as much as possible. With that said, this is an iOS device and not Android, so there is some flexibility to adjust and cater to the iOS environment. This is not necessarily the best example of that, but it is a factor in the decisions we make as a community. 

Once again, you are an active and a contributing member, and I really encourage you to challenge my decisions and be a part of this wonderful process as we make an awesome framework for everyone. 

All the best :)

**Pull request closed without merging**